dashboard/app/reporting_email.go

```diff
@@ -102,6 +106,113 @@ func (cfg *EmailConfig) Validate() error {
 	return nil
 }
 
+// handleCoverageReports sends reports for the previous period.
+// Assuming it is called June 15, the monthly report will cover April-May diff.
+func handleCoverageReports(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	curHostPort := r.URL.Host
+	targetDate := civil.DateOf(timeNow(ctx)).AddMonths(-1)
+	periods, err := coveragedb.GenNPeriodsTill(2, targetDate, "month")
+	if err != nil {
+		msg := fmt.Sprintf("error generating coverage report: %s", err.Error())
+		log.Errorf(ctx, "%s", msg)
+		http.Error(w, "%s: %w", http.StatusBadRequest)
```

Comment:
To avoid duplicating the error handling policy on every path, it's good to add a wrapper function that will have a single copy of the error handling policy.
Maybe we need some middleware for all these cron handlers to do that.

Reply:
Yeah.. we duplicate a lot.... Good point.

Reply:
+1 to generalize it for the cron handlers. Do you mind to ignore it here?

Reply:
re cron - yes